types/node/v20/test/buffer.ts

```diff
@@ -368,6 +365,26 @@ b.fill("a").fill("b");
     b = a.readBigUint64BE(123);
 }
 
+{
+    const buf = Buffer.allocUnsafe(5);
+    let result: Buffer;
+    result = buf.fill("a");
+    result = buf.fill("aazz", "hex");
+    result = buf.fill("aazz", 1, "hex");
+    result = buf.fill("aazz", 1, 2, "hex");
+
+    result = buf.fill(1);
+    result = buf.fill(1234, "hex");
+    result = buf.fill(1234, 1, "hex");
+    result = buf.fill(1234, 1, 2, "hex");
+
+    const target = Buffer.allocUnsafe(0);
+    result = buf.fill(target);
+    result = buf.fill(target, "hex");
+    result = buf.fill(target, 1, "hex");
+    result = buf.fill(target, 1, 2, "hex");
```

Comment:
The `encoding` parameter is ignored when passing a non-string fill value, so we should not be testing these cases.

```suggestion
    result = buf.fill(255);
    result = buf.fill(255, 1);
    result = buf.fill(255, 1, 2);

    const target = Buffer.allocUnsafe(0);
    result = buf.fill(target);
    result = buf.fill(target, 1);
    result = buf.fill(target, 1, 2);
```

Reply:
Thanks! I removed in [73f9a66](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73143/commits/73f9a665a84a0a32571f898c580e606980ad3167).

Reply:
I see the logic, but this disallows passing a fill value of type `Buffer | string`, which is a very common type union. We've come up against this issue before.

The easiest thing to do would be to leave this as you had it previously. Altering this would be a breaking change.

Reply:
Thanks for the clarification — that makes sense.
I'll revert the change and keep the previous definition as suggested.
Appreciate your context and guidance!